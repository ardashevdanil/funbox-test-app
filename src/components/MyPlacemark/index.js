import PropTypes from 'prop-types';
import React from 'react';
import { Placemark } from 'react-yandex-maps';

class MyPlacemark extends React.Component {
  constructor(props) {
    super(props);
    this.handleDrag=this.handleDrag.bind(this);
  }

  handleDrag(e) {
    this.props.onDrag(e, this.props.index);
  }

  render() {
    return (
      <Placemark
        onDrag={this.handleDrag}
        geometry={{
          coordinates: this.props.coords,
          type: 'Point'
        }}
        properties={{
          balloonContent: this.props.value,
          iconContent: this.props.index + 1
        }}
        options={{
          draggable: true,
          preset: 'islands#greenCircleIcon'
        }}
      />
    );
  }
}

MyPlacemark.propTypes = {
  onDrag: PropTypes.func.isRequired,
  coords: PropTypes.arrayOf(PropTypes.number).isRequired,
  value: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
}

export default MyPlacemark;