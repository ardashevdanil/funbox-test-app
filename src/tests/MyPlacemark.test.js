import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

import MyPlacemark from '../components/MyPlacemark';

Enzyme.configure({ adapter: new Adapter() });

describe('test MyPlacemark', () => {
  const onDrag = jest.fn();
	const component = Enzyme.shallow(
    <MyPlacemark
      key={1}
      onDrag={onDrag}
      coords={[1, 1]}
      value='value'
      index={1}
    />
  );

  test('renders correctly', () => {
    expect(component.debug()).toMatchSnapshot();
  });

  test('calls onDrag after drag', () => {
    component.find('Placemark').simulate('drag');
    expect(onDrag.mock.calls.length).toBe(1);
    expect(onDrag.mock.calls[0][1]).toBe(1);
  });
});