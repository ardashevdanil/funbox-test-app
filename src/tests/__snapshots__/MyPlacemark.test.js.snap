// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MyPlacemark renders correctly 1`] = `"<Placemark onDrag={[Function: bound handleDrag]} geometry={{...}} properties={{...}} options={{...}} />"`;
